src/support/bitset.h

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright 2023 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef wasm_support_bitset_h
+#define wasm_support_bitset_h
+
+#include <cassert>
+#include <cstddef>
+#include <iterator>
+#include <vector>
+
+namespace wasm {
+
+// Represent a set of integers in [0, N) with a bitvector of size N, where a 1
```

Comment:
```suggestion
// Represent a set of integers 0 <= k < N with a bitvector of size N, where a 1
```

Reply:
Because the notation `[0, N)` is used in math to imply a half-open interval of real numbers. It's technically fine either way though.

Reply:
Oh, I didn't know the interval notation typically implied real numbers. Will update.

Reply:
Link for the curious: https://en.wikipedia.org/wiki/Interval_(mathematics)